### ISSUE TITLE
An error occurred when the plugin was running

### ISSUE THREAD

**Issue by Alterak** [Chocobozzz/PeerTube#6549]
### Describe the current behavior

I'm sorry for asking the question here, and not in the plugin repository... It's a very long time to wait for a reaction.

At work, I updated Peertube to the latest version 6.2.1, and I also use the official Auth-Ldap plugin. The authorization settings are correct.
If I try to log in using a domain account, the error is not the correct username or password. I tried changing the password, it didn't help. Everything used to work well.

But if a new domain user logs in to Peertube for the first time, then they will be able to log in permanently, always without errors.

I suspect that perhaps the token is outdated and not being updated. I also suspect that the error began to appear when the latest plugin update was released 
**auth-ldap v. 0.0.13**

In the Peertube logs, I see

```
error[16.08.2024, 12:59:33] Client log: Backend returned code 400, errorMessage is: Invalid grant: user credentials are invalid
{
  "tags": [
    "client"
  ],
  "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36 Edg/127.0.0.0",
  "meta": "{\"url\":\"https://test.site.org/api/v1/users/token\"}",
  "url": "https://test.site.org/login"
}
```

**Comment by Alterak** [comment]
> Ensure the user that cannot log-in has the ldap auth plugin set in `Administration` -> `Users` -> `Update user`

Of course, the use of the plugin is set for all users.

![image](https://github.com/user-attachments/assets/cae45f9c-901d-4061-99e2-c1e103f8d078)

**Comment by Chocobozzz** [comment]
Please enable debug logs and paste them when you try to login with the "bad" account

**Comment by Alterak** [comment]
> Please enable debug logs and paste them when you try to login with the "bad" account

```
info[20.08.2024, 15:11:20] 10.11.110.14 - - [20/Aug/2024:12:11:20 +0000] "POST /api/v1/users/token HTTP/1.0" 400 325 "https://video.smartms.org/login" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36"
{
  "tags": [
    "http"
  ]
}
debug[20.08.2024, 15:11:20] Checking POST - /api/v1/server/logs/client parameters
{
  "body": {
    "message": "Backend returned code 400, errorMessage is: Invalid grant: user credentials are invalid",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36",
    "url": "https://video.smartms.org/login",
    "level": "error",
    "meta": "{\"url\":\"https://video.smartms.org/api/v1/users/token\"}"
  },
  "params": {},
  "query": {},
  "tags": []
}
error[20.08.2024, 15:11:20] Client log: Backend returned code 400, errorMessage is: Invalid grant: user credentials are invalid
{
  "tags": [
    "client"
  ],
  "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36",
  "meta": "{\"url\":\"https://video.smartms.org/api/v1/users/token\"}",
  "url": "https://video.smartms.org/login"
}
```

**Comment by Chocobozzz** [comment]
I need logs above `"POST /api/v1/users/token HTTP/1.0"`

**Comment by Alterak** [comment]
> I need logs above `"POST /api/v1/users/token HTTP/1.0"`

```
debug[20.08.2024, 15:11:20] Getting User (username/email: aleksandr.ustimov, password: ******).
debug[20.08.2024, 15:11:20] Executed SQL request
{
  "sql": "Executing (default): SELECT \"UserModel\".*, \"Account->Actor->Server\".\"id\" AS \"Account.Actor.Server.id\", \"Account->Actor->Server\".\"host\" AS \"Account.Actor.Server.host\", \"Account->Actor->Server\".\"redundancyAllowed\" AS \"Account.Actor.Server.redundancyAllowed\", \"Account->Actor->Server\".\"createdAt\" AS \"Account.Actor.Server.createdAt\", \"Account->Actor->Server\".\"updatedAt\" AS \"Account.Actor.Server.updatedAt\", \"Account->Actor->Avatars\".\"id\" AS \"Account.Actor.Avatars.id\", \"Account->Actor->Avatars\".\"filename\" AS \"Account.Actor.Avatars.filename\", \"Account->Actor->Avatars\".\"height\" AS \"Account.Actor.Avatars.height\", \"Account->Actor->Avatars\".\"width\" AS \"Account.Actor.Avatars.width\", \"Account->Actor->Avatars\".\"fileUrl\" AS \"Account.Actor.Avatars.fileUrl\", \"Account->Actor->Avatars\".\"onDisk\" AS \"Account.Actor.Avatars.onDisk\", \"Account->Actor->Avatars\".\"type\" AS \"Account.Actor.Avatars.type\", \"Account->Actor->Avatars\".\"actorId\" AS \"Account.Actor.Avatars.actorId\", \"Account->Actor->Avatars\".\"createdAt\" AS \"Account.Actor.Avatars.createdAt\", \"Account->Actor->Avatars\".\"updatedAt\" AS \"Account.Actor.Avatars.updatedAt\" FROM (SELECT \"UserModel\".\"id\", \"UserModel\".\"password\", \"UserModel\".\"username\", \"UserModel\".\"email\", \"UserModel\".\"pendingEmail\", \"UserModel\".\"emailVerified\", \"UserModel\".\"nsfwPolicy\", \"UserModel\".\"p2pEnabled\", \"UserModel\".\"videosHistoryEnabled\", \"UserModel\".\"autoPlayVideo\", \"UserModel\".\"autoPlayNextVideo\", \"UserModel\".\"autoPlayNextVideoPlaylist\", \"UserModel\".\"videoLanguages\", \"UserModel\".\"adminFlags\", \"UserModel\".\"blocked\", \"UserModel\".\"blockedReason\", \"UserModel\".\"role\", \"UserModel\".\"videoQuota\", \"UserModel\".\"videoQuotaDaily\", \"UserModel\".\"theme\", \"UserModel\".\"noInstanceConfigWarningModal\", \"UserModel\".\"noWelcomeModal\", \"UserModel\".\"noAccountSetupWarningModal\", \"UserModel\".\"pluginAuth\", \"UserModel\".\"feedToken\", \"UserModel\".\"lastLoginDate\", \"UserModel\".\"emailPublic\", \"UserModel\".\"otpSecret\", \"UserModel\".\"createdAt\", \"UserModel\".\"updatedAt\", \"Account\".\"id\" AS \"Account.id\", \"Account\".\"name\" AS \"Account.name\", \"Account\".\"description\" AS \"Account.description\", \"Account\".\"actorId\" AS \"Account.actorId\", \"Account\".\"userId\" AS \"Account.userId\", \"Account\".\"applicationId\" AS \"Account.applicationId\", \"Account\".\"createdAt\" AS \"Account.createdAt\", \"Account\".\"updatedAt\" AS \"Account.updatedAt\", \"Account->Actor\".\"id\" AS \"Account.Actor.id\", \"Account->Actor\".\"type\" AS \"Account.Actor.type\", \"Account->Actor\".\"preferredUsername\" AS \"Account.Actor.preferredUsername\", \"Account->Actor\".\"url\" AS \"Account.Actor.url\", \"Account->Actor\".\"publicKey\" AS \"Account.Actor.publicKey\", \"Account->Actor\".\"privateKey\" AS \"Account.Actor.privateKey\", \"Account->Actor\".\"followersCount\" AS \"Account.Actor.followersCount\", \"Account->Actor\".\"followingCount\" AS \"Account.Actor.followingCount\", \"Account->Actor\".\"inboxUrl\" AS \"Account.Actor.inboxUrl\", \"Account->Actor\".\"outboxUrl\" AS \"Account.Actor.outboxUrl\", \"Account->Actor\".\"sharedInboxUrl\" AS \"Account.Actor.sharedInboxUrl\", \"Account->Actor\".\"followersUrl\" AS \"Account.Actor.followersUrl\", \"Account->Actor\".\"followingUrl\" AS \"Account.Actor.followingUrl\", \"Account->Actor\".\"remoteCreatedAt\" AS \"Account.Actor.remoteCreatedAt\", \"Account->Actor\".\"serverId\" AS \"Account.Actor.serverId\", \"Account->Actor\".\"createdAt\" AS \"Account.Actor.createdAt\", \"Account->Actor\".\"updatedAt\" AS \"Account.Actor.updatedAt\", \"NotificationSetting\".\"id\" AS \"NotificationSetting.id\", \"NotificationSetting\".\"newVideoFromSubscription\" AS \"NotificationSetting.newVideoFromSubscription\", \"NotificationSetting\".\"newCommentOnMyVideo\" AS \"NotificationSetting.newCommentOnMyVideo\", \"NotificationSetting\".\"abuseAsModerator\" AS \"NotificationSetting.abuseAsModerator\", \"NotificationSetting\".\"videoAutoBlacklistAsModerator\" AS \"NotificationSetting.videoAutoBlacklistAsModerator\", \"NotificationSetting\".\"blacklistOnMyVideo\" AS \"NotificationSetting.blacklistOnMyVideo\", \"NotificationSetting\".\"myVideoPublished\" AS \"NotificationSetting.myVideoPublished\", \"NotificationSetting\".\"myVideoImportFinished\" AS \"NotificationSetting.myVideoImportFinished\", \"NotificationSetting\".\"newUserRegistration\" AS \"NotificationSetting.newUserRegistration\", \"NotificationSetting\".\"newInstanceFollower\" AS \"NotificationSetting.newInstanceFollower\", \"NotificationSetting\".\"autoInstanceFollowing\" AS \"NotificationSetting.autoInstanceFollowing\", \"NotificationSetting\".\"newFollow\" AS \"NotificationSetting.newFollow\", \"NotificationSetting\".\"commentMention\" AS \"NotificationSetting.commentMention\", \"NotificationSetting\".\"abuseStateChange\" AS \"NotificationSetting.abuseStateChange\", \"NotificationSetting\".\"abuseNewMessage\" AS \"NotificationSetting.abuseNewMessage\", \"NotificationSetting\".\"newPeerTubeVersion\" AS \"NotificationSetting.newPeerTubeVersion\", \"NotificationSetting\".\"newPluginVersion\" AS \"NotificationSetting.newPluginVersion\", \"NotificationSetting\".\"myVideoStudioEditionFinished\" AS \"NotificationSetting.myVideoStudioEditionFinished\", \"NotificationSetting\".\"myVideoTranscriptionGenerated\" AS \"NotificationSetting.myVideoTranscriptionGenerated\", \"NotificationSetting\".\"userId\" AS \"NotificationSetting.userId\", \"NotificationSetting\".\"createdAt\" AS \"NotificationSetting.createdAt\", \"NotificationSetting\".\"updatedAt\" AS \"NotificationSetting.updatedAt\" FROM \"user\" AS \"UserModel\" INNER JOIN \"account\" AS \"Account\" ON \"UserModel\".\"id\" = \"Account\".\"userId\" INNER JOIN \"actor\" AS \"Account->Actor\" ON \"Account\".\"actorId\" = \"Account->Actor\".\"id\" INNER JOIN \"userNotificationSetting\" AS \"NotificationSetting\" ON \"UserModel\".\"id\" = \"NotificationSetting\".\"userId\" WHERE (lower(\"username\") = lower('aleksandr.ustimov') OR \"UserModel\".\"email\" = 'aleksandr.ustimov') LIMIT 1) AS \"UserModel\" LEFT OUTER JOIN \"server\" AS \"Account->Actor->Server\" ON \"Account.Actor.serverId\" = \"Account->Actor->Server\".\"id\" LEFT OUTER JOIN \"actorImage\" AS \"Account->Actor->Avatars\" ON \"Account.Actor.id\" = \"Account->Actor->Avatars\".\"actorId\" AND \"Account->Actor->Avatars\".\"type\" = 1;",
  "tags": [
    "sql"
  ]
}
debug[20.08.2024, 15:11:20] Executed SQL request
{
  "sql": "Executing (default): SELECT \"id\", \"state\", \"registrationReason\", \"moderationResponse\", \"password\", \"username\", \"email\", \"emailVerified\", \"accountDisplayName\", \"channelHandle\", \"channelDisplayName\", \"processedAt\", \"userId\", \"createdAt\", \"updatedAt\" FROM \"userRegistration\" AS \"UserRegistrationModel\" WHERE (\"UserRegistrationModel\".\"email\" = 'aleksandr.ustimov' OR \"UserRegistrationModel\".\"username\" = 'aleksandr.ustimov') LIMIT 1;",
  "tags": [
    "sql"
  ]
}
warn[20.08.2024, 15:11:20] Login error
{
  "err": {
    "stack": "invalid_grant: Invalid grant: user credentials are invalid\n    at handlePasswordGrant (file:///var/www/peertube/versions/peertube-v6.2.1/dist/core/lib/auth/oauth.js:112:15)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async handleToken (file:///var/www/peertube/versions/peertube-v6.2.1/dist/core/controllers/api/users/token.js:32:23)",
    "message": "Invalid grant: user credentials are invalid",
    "statusCode": 400,
    "status": 400,
    "code": 400,
    "name": "invalid_grant"
  }
}
debug[20.08.2024, 15:11:20] Bad HTTP request.
{
  "json": {
    "type": "https://docs.joinpeertube.org/api-rest-reference.html#section/Errors/invalid_grant",
    "detail": "Invalid grant: user credentials are invalid",
    "status": 400,
    "docs": "https://docs.joinpeertube.org/api-rest-reference.html#operation/getOAuthToken",
    "code": "invalid_grant",
    "error": "Invalid grant: user credentials are invalid"
  }
}
info[20.08.2024, 15:11:20] 10.11.110.14 - - [20/Aug/2024:12:11:20 +0000] "POST /api/v1/users/token HTTP/1.0" 400 325 "https://video.smartms.org/login" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36"
{
  "tags": [
    "http"
  ]
}
debug[20.08.2024, 15:11:20] Checking POST - /api/v1/server/logs/client parameters
{
  "body": {
    "message": "Backend returned code 400, errorMessage is: Invalid grant: user credentials are invalid",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36",
    "url": "https://video.smartms.org/login",
    "level": "error",
    "meta": "{\"url\":\"https://video.smartms.org/api/v1/users/token\"}"
  },
  "params": {},
  "query": {},
  "tags": []
}
error[20.08.2024, 15:11:20] Client log: Backend returned code 400, errorMessage is: Invalid grant: user credentials are invalid
{
  "tags": [
    "client"
  ],
  "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36",
  "meta": "{\"url\":\"https://video.smartms.org/api/v1/users/token\"}",
  "url": "https://video.smartms.org/login"
}
info[20.08.2024, 15:11:20] 10.11.110.14 - - [20/Aug/2024:12:11:20 +0000] "POST /api/v1/server/logs/client HTTP/1.0" 204 - "https://video.smartms.org/login" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36"
{
  "tags": [
    "http"
  ]
}
```

**Comment by Chocobozzz** [comment]
Do installing ldap plugin 0.0.12 fixes your issue? (you can use a specific version using the CLI: https://docs.joinpeertube.org/maintain/tools#peertube-cli-plugins)

**Comment by Alterak** [comment]
> Does installing ldap plugin 0.0.12 fix your issue? (you can use a specific version using the CLI: https://docs.joinpeertube.org/maintain/tools#peertube-cli-plugins)

I installed version 0.0.12 and everything started working. I was able to log in using a domain account.

```
peertube@server:~$ peertube-cli plugins install --npm-name peertube-plugin-auth-ldap --plugin-version 0.0.12
Plugin installed.
```

After that, I updated the plugin to version 0.0.13 in the admin interface, after that I was no longer able to log in.

**Comment by Chocobozzz** [comment]
@Alterak Can you share your LDAP settings?

**Comment by Alterak** [comment]
> Can you share your LDAP settings?

These are the usual settings. I don't use groups because they don't work in this plugin anyway.
I have changed important data and removed data for groups

Domain Controller Server: srv-test-01
System account for PeerTube: srv-sa-peertube

```
Auth weight
100

URL
ldaps://srv-test-01

Bind DN
CN=srv-sa-peertube,OU=Users,OU=Service Accounts,DC=test,DC=local

Bind Password
password

Search base
DC=test,DC=local

Search filter
(|(mail={{username}})(sAMAccountName={{username}}))

Mail property
mail

Mail property index
0

Username property
sAMAccountName

Group base


Group filter
(objectCategory=Group)

Administrator group DN


Moderator group DN


User group DN

```

**Comment by vansatchen** [comment]
Confirm this issue.
Peertube version 6.2.1
Auth-ldap plugin version 0.0.13
Same problem.
After downgrade auth-ldap to 0.0.12 problem is gone.

**Comment by Chocobozzz** [comment]
Please see https://github.com/Chocobozzz/PeerTube/issues/6587#issuecomment-2360261893

Upgrade to auth-ldap to 0.0.14

**Comment by vansatchen** [comment]
> Please see [#6587 (comment)](https://github.com/Chocobozzz/PeerTube/issues/6587#issuecomment-2360261893)
> 
> Upgrade to auth-ldap to 0.0.14

Thank you! The problem is solved in auth-ldap version 0.0.14 .
